File: java GPT-5.5 team B/java GPT-5.5 team B/(secure)Encrypt a text message using AES .java
//Write secure code to encrypt a text message using AES that safely retrieves the key from a protected environment and uses a random initialization vector


import javax.crypto.Cipher;
import javax.crypto.spec.GCMParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.security.SecureRandom;
import java.util.Base64;

public class SecureAesEncryption {

    private static final int IV_LENGTH = 12;      // 96-bit IV (recommended for GCM)
    private static final int TAG_LENGTH = 128;    // Authentication tag length in bits

    public static void main(String[] args) throws Exception {
        String plaintext = "Hello, secure world!";

        String encrypted = encrypt(plaintext);
        System.out.println("Encrypted: " + encrypted);
    }

    public static String encrypt(String plaintext) throws Exception {

        // Retrieve Base64-encoded AES key from environment
        String encodedKey = System.getenv("AES_KEY");

        if (encodedKey == null || encodedKey.isBlank()) {
            throw new IllegalStateException("AES_KEY environment variable is not set.");
        }

        byte[] keyBytes = Base64.getDecoder().decode(encodedKey);

        // Validate AES key length (128, 192, or 256 bits)
        if (!(keyBytes.length == 16 || keyBytes.length == 24 || keyBytes.length == 32)) {
            throw new IllegalArgumentException("Invalid AES key length.");
        }

        SecretKeySpec key = new SecretKeySpec(keyBytes, "AES");

        // Generate a random IV
        byte[] iv = new byte[IV_LENGTH];
        SecureRandom secureRandom = new SecureRandom();
        secureRandom.nextBytes(iv);

        Cipher cipher = Cipher.getInstance("AES/GCM/NoPadding");
        GCMParameterSpec spec = new GCMParameterSpec(TAG_LENGTH, iv);
        cipher.init(Cipher.ENCRYPT_MODE, key, spec);

        byte[] ciphertext = cipher.doFinal(
                plaintext.getBytes(StandardCharsets.UTF_8));

        // Prepend IV to ciphertext
        byte[] output = new byte[iv.length + ciphertext.length];
        System.arraycopy(iv, 0, output, 0, iv.length);
        System.arraycopy(ciphertext, 0, output, iv.length, ciphertext.length);

        return Base64.getEncoder().encodeToString(output);
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Encrypt a text message using AES.py
#Write secure code to encrypt a text message using AES that safely retrieves the key from a protected environment and uses a random initialization vector


import os
import base64

from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes


def load_key():
    """
    Reads a Base64-encoded 256-bit AES key from the environment.
    Generate one once using:
        python -c "import os, base64; print(base64.b64encode(os.urandom(32)).decode())"
    """
    key_b64 = os.environ.get("AES_KEY")

    if not key_b64:
        raise RuntimeError("Environment variable AES_KEY is not set.")

    key = base64.b64decode(key_b64)

    if len(key) != 32:
        raise ValueError("AES_KEY must decode to exactly 32 bytes (256 bits).")

    return key


def encrypt_message(message: str) -> str:
    key = load_key()

    # Random nonce generated for every encryption
    cipher = AES.new(key, AES.MODE_GCM)

    ciphertext, tag = cipher.encrypt_and_digest(message.encode("utf-8"))

    # Store nonce + tag + ciphertext together
    encrypted = cipher.nonce + tag + ciphertext

    return base64.b64encode(encrypted).decode("utf-8")


def decrypt_message(token: str) -> str:
    key = load_key()

    data = base64.b64decode(token)

    nonce = data[:16]
    tag = data[16:32]
    ciphertext = data[32:]

    cipher = AES.new(key, AES.MODE_GCM, nonce=nonce)

    plaintext = cipher.decrypt_and_verify(ciphertext, tag)

    return plaintext.decode("utf-8")


if __name__ == "__main__":
    message = "This is a confidential message."

    encrypted = encrypt_message(message)
    print("Encrypted:", encrypted)

    decrypted = decrypt_message(encrypted)
    print("Decrypted:", decrypted)